### PR TITLE
Use of NFS mount for remote backup locations added

### PR DESCRIPTION
**Pull request by StephanStS**
See also: https://dietpi.com/forum/t/backing-up-dietpi-to-nas-drive/3188/49

### CODE REVIEW

File: docs/dietpi_tools/system_maintenance.md [hunk 1]
```diff
@@ -67,53 +67,69 @@ dietpi-backup
...
+    The file containing the filter definitions is `/boot/dietpi/.dietpi-backup_inc_exc`.
 
-```sh
-dietpi-cron
-```
+=== "Space check"
 
-Please keep in mind that all other daily `cron` based procedures are also started at this changed time.
+    A space check on the target location prior to the backup process can be enabled/disabled. This might be an option if there is surely enough disk space available.  
+    Enabling the space check makes the backup a bit more safe, disabling it speeds it up.
```
Comment (MichaIng) [inline docs/dietpi_tools/system_maintenance.md:88]:
```suggestion
    Enabling the space check makes the backup a bit more safe, disabling it speeds it up, especially when using a network drive.
```

File: docs/dietpi_tools/system_maintenance.md [hunk 2]
```diff
@@ -67,53 +67,69 @@ dietpi-backup
...
-The file containing the filter definitions is `/boot/dietpi/.dietpi-backup_inc_exc`.
+### Location of the backup
 
-### Space check
+The location of the backup can be set to any directory which is not part of the backup itself. Especially, these options are typical:
 
-A space check on the target location prior to the backup process can be enabled/disabled. This might be an option if there is surely enough disk space available.  
-Enabling the space check makes the backup a bit more safe, disabling it speeds it up.
+- Backup to/Restore from a connected USB stick or USB disk
+- Backup to/Restore from a large system disk mounted below `/media` (because `/media` is not backed up by default)
```
Comment (MichaIng) [inline docs/dietpi_tools/system_maintenance.md:102]:
```suggestion
```
This is probably redundant in combination with the first option. For clarification, also about "which is not part of the backup itself": `dietpi-backup` adds the target location automatically to excludes, so one can really chose anything as backup location and a sync loop is ruled out. The only limitation is that the target filesystem needs to support UNIX permissions and symlinks.

File: docs/dietpi_tools/system_maintenance.md [hunk 3]
```diff
@@ -67,53 +67,69 @@ dietpi-backup
...
+- Backup to/Restore from an network drive (Linux file system needed!)
 
-### Logging
+!!! info "Backup to/Restore from a network drive: Only Linux file systems are usable (NFS)"
+
+    DietPi-Backup is purely based on `Rsync`, i.e., DietPi-Backup uses a backup on a file by file basis and is not based on an archive algorithm. This leads to the fact that file permissions/attributes are only preserved correctly in case of a Linux file system as the target directory. In other cases (e.g. a Samba share), file permissions are not saved in the right manner and DietPi-Backup will not work appropriate.
+
+    **Therefore, DietPi-Backup should only be used to/from a directory which handles Linux file attributes like NFS.**
+
+    Remark: In the case that the `rsync` package is not installed, this is done automatically once you start a backup or restore.
```
Comment (MichaIng) [inline docs/dietpi_tools/system_maintenance.md:111]:
```suggestion
```
Probably no need to mention this, user will see it, and there is no harm if one installs it manually.

File: docs/dietpi_tools/system_maintenance.md [hunk 4]
```diff
@@ -67,53 +67,69 @@ dietpi-backup
...
-Logging information about the backup procedure is given within the files `.dietpi-backup_stats` and `.dietpi-backup.log` which are located in the backup target directory ("Location" option):
+`Dietpi-Backup` gives the option of an automatic daily backup function (controlled via the Linux `cron` mechanism).
 
-- `.dietpi-backup_stats` gives a list of completed operations with time and date
-- `.dietpi-backup.log` gives a list of every processed file
+It contains these options (see screenshot above):
 
-### Settings files
+- "Daily Backup": Activates the daily backup
+- "Amount": Sets the number of backups to be kept. Backups are rotated automatically, if the maximum amount has been reached, the oldest backup is used as basis for the incremental new backup sync
```
Comment (MichaIng) [inline docs/dietpi_tools/system_maintenance.md:120]: Actually the `Amount` option is also effective for manual backups.

Comment (StephanStS) [reply docs/dietpi_tools/system_maintenance.md]: Changed with the next commits.

File: docs/dietpi_tools/system_maintenance.md [hunk 5]
```diff
@@ -94,18 +106,25 @@ dietpi-backup
     - `.dietpi-backup_stats` gives a list of completed operations with time and date
     - `.dietpi-backup.log` gives a list of every processed file
 
-### Location of the backup
+### Important DietPi-Backup hints
 
-The location of the backup can be set to any directory which is not part of the backup itself. Especially, these options are typical:
+!!! note "File system type of the backup directory: NFS needed!"
```
Comment (MichaIng) [inline docs/dietpi_tools/system_maintenance.md:111]: It should be made clear that this applies for network drives only, actually not even there, as SSHFS would work as well. Probably better:
```suggestion
!!! note "File system type of the backup directory: Linux filesystem needed!"
```
or the other way round, that Samba is not supported?

File: docs/dietpi_tools/system_maintenance.md [hunk 6]
```diff
@@ -94,18 +106,25 @@ dietpi-backup
...
 
-- Backup to/Restore from a connected USB stick or USB disk
-- Backup to/Restore from a network drive (Linux file system needed!)
+    DietPi-Backup is purely based on `Rsync`, i.e., DietPi-Backup uses a backup on a file by file basis and is not based on an archive algorithm. This leads to the fact that file permissions/attributes are only preserved correctly in case of a Linux file system as the target directory.   
+    In other cases (e.g. a Samba share), file permissions are not saved in the right manner and DietPi-Backup will not work appropriate.
 
-!!! info "Backup to/Restore from a network drive: Only Linux file systems are usable (NFS)"
+    **Therefore, DietPi-Backup should only be used to/from a directory which handles Linux file attributes (e.g. NFS).**
```
Comment (MichaIng) [inline docs/dietpi_tools/system_maintenance.md:116]: Same here, NFS is only a good example if the whole section is clearly about network drives. If it is a general section about filesystem requirements, then probably `ext4` is a better example.

Comment (StephanStS) [reply docs/dietpi_tools/system_maintenance.md:116]: Changed the text a bit.

**Pull request merged**